csvy/validators/table_schema.py

```diff
@@ -0,0 +1,454 @@
+    """Validator for the constraints in the Table Schema.
+
+    This class is used to validate the constraints in the Table Schema. It is based on
+    the constraints defined in the Table Schema specification.
+
+    Attributes:
+        required: A boolean indicating if the value is required.
+        unique: A boolean indicating if the value is unique.
+        minimum: The minimum of the value. Applies to types: integer, number, date,
+            time, datetime, year, yearmonth.
+        maximum: The maximum value. Applies to types: integer, number, date,
+            time, datetime, year, yearmonth.
+        minLength: The minimum length of the field. Applies to collections (string,
+            array, object).
+        maxLength: The maximum length of the field. Applies to collections (string,
+            array, object).
+        pattern: A regular expression pattern that the value must match. Applies to
+            types: string.
+        enum: A list of possible values for the field.
+
+    """
+
+    required: bool | None = Field(None)
+    unique: bool | None = Field(None)
+    minimum: int | float | None = Field(None)
+    maximum: int | float | None = Field(None)
+    minLength: int | None = Field(None)
+    maxLength: int | None = Field(None)
+    pattern: str | None = Field(None)
+    enum: list[Any] | None = Field(None)
+
+
+class CommonValidator(BaseModel):
+    """Validator for the columns in the Table Schema.
+
+    This class is used to validate the columns in the Table Schema. It is based on the
+    columns defined in the Table Schema specification.
+
+    Attributes:
+        name: The name of the column.
+        title: A nicer human readable label or title for the field.
+        example: An example value for the field.
+        description: A description for the field.
+        constraints: A dictionary of constraints for the field.
+
+    """
+
+    name: str = Field(..., description="Column name.")
+    title: str | None = Field(
+        None, description="A nicer human readable label or title for the field."
+    )
+    example: str | None = Field(None, description="An example value for the field.")
+    description: str | None = Field(None, description="A description for the field.")
+    constraints: ConstraintsValidator = Field(
+        ConstraintsValidator(), description="A dictionary of constraints for the field."
+    )
+
+    def model_dump(self, *args, **kwargs) -> dict[str, Any]:
+        """Dump the model to a dictionary.
+
+        This method dumps the model to a dictionary. It sets exclude_unset to True and
+        by_alias to True, so that only the attributes that were set are included in the
+        dictionary and their aliases are always used.
+
+        Finally, it converts the attributes that are Enum instances to their values.
+
+        Returns:
+            A dictionary with the model attributes.
+
+        """
+        kwargs["exclude_unset"] = True
+        kwargs["by_alias"] = True
```

Comment:
Overwriting `kwargs` like this feels wrong. It means someone who is familiar with Pydantic might want to try `validator.model_dump(exclude_unset=False)` and not get the result they're expecting. I think it's better to either rename this method so the regular `model_dump` is still available, or include something like
```suggestion
        if "exclude_unset" not in kwargs:
              kwargs["exclude_unset"] = True
        if "by_alias" not in kwargs:
              kwargs["by_alias"] = True
```

Reply:
Good point. What about the following, instead, which feels more concise:
```suggestion
        kwargs["exclude_unset"] = kwargs.get("exclude_unset", True)
        kwargs["by_alias"] = kwargs.get("by_alias", True)
```

Reply:
Seems sensible!